mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp

```diff
@@ -257,21 +290,24 @@ DenseElementsAttr transpose(ElementsAttr attr, ShapedType inputType,
     case 1:
       return transposeType<bool>(attr, inputType, outputType, permValues);
     case 8:
-      return transposeType<int8_t>(attr, inputType, outputType, permValues);
+      return transposeTypeRaw<int8_t>(attr, inputType, outputType, permValues);
     case 16:
-      return transposeType<int16_t>(attr, inputType, outputType, permValues);
+      return transposeTypeRaw<int16_t>(attr, inputType, outputType, permValues);
     case 32:
-      return transposeType<int32_t>(attr, inputType, outputType, permValues);
+      return transposeTypeRaw<int32_t>(attr, inputType, outputType, permValues);
     case 64:
-      return transposeType<int64_t>(attr, inputType, outputType, permValues);
+      return transposeTypeRaw<int64_t>(attr, inputType, outputType, permValues);
     default:
       return transposeType<APInt>(attr, inputType, outputType, permValues);
     }
   }
 
   // Handle possible float types
   if (baseType.isF32()) {
-    return transposeType<float>(attr, inputType, outputType, permValues);
+    return transposeTypeRaw<float>(attr, inputType, outputType, permValues);
```

Comment:
I'd have a small preference for `transpose` was done based on `uint` containers with the right storage size for the element rather than having any knowledge of what the bits mean for plain old data types. But this is more a nit as there should be little other effect.